internal/http/services/owncloud/ocs/conversions/permissions.go

```diff
@@ -38,21 +38,30 @@ const (
 	PermissionDelete
 	// PermissionShare grants share permissions on a resource
 	PermissionShare
+	// PermissionDeny grants permissions to deny access on a resource
+	// The recipient of the resource will then have PermissionNone.
+	PermissionDeny
+	// PermissionNone grants no permissions on a resource
+	PermissionNone
+	// PermissionMax is to be used within value range checks
+	PermissionMax Permissions = (1 << (iota - 1)) - 1
 	// PermissionAll grants all permissions on a resource
-	PermissionAll Permissions = (1 << (iota - 1)) - 1
+	PermissionAll = PermissionMax - PermissionNone
+	// PermissionMin is to be used within value range checks
+	PermissionMin = PermissionRead
```

Comment:
We should not change the bitmask. We should use roles instead. The permissions bitmask is considered deprecated and will not work in spaces.

Reply:
That's fine, this PR is not expected to work with spaces, only for the master branch.

Reply:
Ok but the testsuite will never pass on this. The permissions bitmask ends with `31` everything above that will be considered as "invalid permission"

Reply:
Yeah we found that earlier, I think we will add those tests in the expected failures files for time being as this will (are) fixed in the edge branch.